lib/presentation/redux/reducers/shared_space_document_reducer.dart

```diff
@@ -0,0 +1,76 @@
+// for more details.
+// This program is distributed in the hope that it will be useful, but WITHOUT ANY
+// WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+// FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for
+// more details.
+// You should have received a copy of the GNU Affero General Public License and its
+// applicable Additional Terms for LinShare along with this program. If not, see
+// <http://www.gnu.org/licenses/> for the GNU Affero General Public License version
+//  3 and <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf> for
+//  the Additional Terms applicable to LinShare software.
+//
+
+import 'package:domain/domain.dart';
+import 'package:linshare_flutter_app/presentation/redux/actions/shared_space_document_action.dart';
+import 'package:linshare_flutter_app/presentation/redux/states/shared_space_document_state.dart';
+import 'package:redux/redux.dart';
+
+final sharedSpaceDocumentReducer = combineReducers<SharedSpaceDocumentState>([
+  TypedReducer<SharedSpaceDocumentState, StartSharedSpaceDocumentLoadingAction>((SharedSpaceDocumentState state, _) => state.startLoadingState()),
+  TypedReducer<SharedSpaceDocumentState, SharedSpaceDocumentAction>((SharedSpaceDocumentState state, SharedSpaceDocumentAction action) => state.sendViewState(viewState: action.viewState)),
+  TypedReducer<SharedSpaceDocumentState, CleanSharedSpaceDocumentStateAction>((SharedSpaceDocumentState state, _) => state.clearViewState()),
+  TypedReducer<SharedSpaceDocumentState, UpdateSharedSpaceDocumentArgumentsAction>((SharedSpaceDocumentState state, UpdateSharedSpaceDocumentArgumentsAction action) =>
+    state.setSharedSpaceDocumentArgument(newArguments: action.documentArguments)),
+  TypedReducer<SharedSpaceDocumentState, SharedSpaceDocumentGetSorterAndAllWorkGroupNodeAction>((SharedSpaceDocumentState state, SharedSpaceDocumentGetSorterAndAllWorkGroupNodeAction action) =>
+    state.setSharedSpaceDocument(
+      viewState: action.viewState,
+      newWorkGroupNodeList: action.viewState.fold(
+        (failure) => [],
+        (success) => (success is GetChildNodesViewState) ? success.workGroupNodes : []),
+      newSorter: action.sorter
+    )),
+  TypedReducer<SharedSpaceDocumentState, SharedSpaceDocumentGetAllWorkGroupNodeAction>((SharedSpaceDocumentState state, SharedSpaceDocumentGetAllWorkGroupNodeAction action) =>
+    state.setSharedSpaceDocument(
+      viewState: action.viewState,
+      newWorkGroupNodeList: action.viewState.fold(
+        (failure) => [],
+        (success) => (success is GetChildNodesViewState) ? success.workGroupNodes : [])
+    )),
+  TypedReducer<SharedSpaceDocumentState, SharedSpaceDocumentSortWorkGroupNodeAction>((SharedSpaceDocumentState state, SharedSpaceDocumentSortWorkGroupNodeAction action) =>
+    state.setSharedSpaceDocument(
+      newWorkGroupNodeList: action.workGroupNodes,
+      newSorter: action.sorter
+    )),
```

Comment:
when you split line, why not using `{ ... }` instead of `=>`

Reply:
same for other

Reply:
I think you are too careful. But for your pleasure I will fix it.